cmds/tui.c

```diff
@@ -2032,6 +2008,48 @@ static bool tui_window_expand(struct tui_window *win)
 	return win->ops->expand(win, win->curr);
 }
 
+/* move to the previous sibling */
+static bool tui_window_move_prev(struct tui_window *win)
+{
+	struct tui_graph_node *curr = win->curr;
+	void *prev = win->ops->sibling_prev(win, curr);
+
+	if (prev == NULL)
+		return false;
+
+	/* fold the current node before moving to the previous sibling */
+	fold_graph_node(curr, true);
+
+	while (win->curr != prev)
+		tui_window_move_up(win);
+
+	/* collapse the current node after moving to the previous sibling */
+	tui_window_collapse(win);
+
+	return true;
+}
```

Comment:
What about this?

```c
static bool tui_window_move_prev(win) {
   prev = win->ops->sibling_prev(win, win->curr);
   parent = win->ops->parent(win, win->curr);

   if (prev == NULL)
      return false;

   if (parent != NULL)
      win->ops->collapse(win, parent);

   while (win->curr != prev)
      tui_window_move_up(win);

   return tui_window_collapse(win);
}
```

Reply:
Hmm.. It makes the time on the left disappears.

Reply:
What about returning `true` instead?

Reply:
Okay, I will test it. By the way, why do we collapse the parent in this change? I think we discussed that we don’t touch other nodes except for prev and next.

Reply:
I just tested but it has another bugs..

Reply:
Maybe you can change `win->ops->collapse()` and `win->ops->expand()` to pass `depth` argument and use it in the generic layer.